pkg/authz/distributedtx/activity.go

```diff
@@ -40,6 +42,29 @@ func (h *ActivityHandler) WriteToSpiceDB(ctx context.Context, input *v1.WriteRel
 	return out, err
 }
 
+// ReadRelationships reads relationships from spicedb and returns any errors.
+func (h *ActivityHandler) ReadRelationships(ctx context.Context, input *v1.ReadRelationshipsRequest) ([]*v1.ReadRelationshipsResponse, error) {
+	failpoints.FailPoint("panicReadSpiceDB")
+	resp, err := h.PermissionClient.ReadRelationships(ctx, input)
+	failpoints.FailPoint("panicSpiceDBReadResp")
+	if err != nil {
+		return nil, err
+	}
+
+	results := make([]*v1.ReadRelationshipsResponse, 0)
+	for {
+		resp, err := resp.Recv()
```

Comment:
`ReadRelationshipResponse` is a paginated response, I don't see that being handled, and accumulating it in memory could cause the process to get OOMKilled